### PR TITLE
Row-level TTL PR 9: TTD test with ttl deduplicator

### PR DESCRIPTION
**Pull request by ableegoldman**
No changes to the ttl implementation, just adding another TTD test to replicate a ttl deduplicator type app. Due to a small discrepency in the way timestamps are resolved in the TestInputTopics created by the TTD, users will need to make sure they specify an initial time for the TTD itself and either (a) pass the same starting time in to the the driver's #createInputTopic method when creating a TestInputTopic, or (b) set  timestamps for each record passed in via the topic's #pipeInput method.

I submitted a fix for this issue in the Kafka Streams TTD, but we'll need to wait for that to be released before the Responsive TTD will work without setting the times as described above.

Also includes some refactoring of test code to reduce duplicate code for common things like custom processors, processor suppliers, and certain topologies we use in multiple places like the deduplicator

PR 1: API https://github.com/responsivedev/responsive-pub/pull/370
PR 2: TtlResolver https://github.com/responsivedev/responsive-pub/pull/371
PR 3: compute minValidTs from TtlResolver https://github.com/responsivedev/responsive-pub/pull/373
PR 4: CassandraFactTable implementation https://github.com/responsivedev/responsive-pub/pull/374
PR 5: extract StateSerdes https://github.com/responsivedev/responsive-pub/pull/375
PR 6: support for ResponsiveTopologyTestDriver https://github.com/responsivedev/responsive-pub/pull/376
PR 7: TopologyTestDriver#advanceWallClockTime https://github.com/responsivedev/responsive-pub/pull/379
PR 8: advance streamTime in #get and add RowLevelTtlIntegrationTest https://github.com/responsivedev/responsive-pub/pull/380

### CODE REVIEW

File: kafka-client/build.gradle.kts [hunk 1]
```diff
@@ -62,6 +62,19 @@ tasks.publish {
     dependsOn(tasks[writeVersionPropertiesFile])
 }
 
+configurations {
```
Comment (ableegoldman) [inline kafka-client/build.gradle.kts:65]: I'll admit I had to ask chatGPT how to access the stuff in the testutils of `kafka-client` from the tests in `responsive-test-utils`, and this was how it said to do it. Not sure it's necessary the right way to do so, if you have a different suggestion

File: kafka-client/src/test/java/dev/responsive/kafka/integration/RowLevelTtlIntegrationTest.java [hunk 2]
```diff
@@ -202,35 +193,13 @@ public void shouldApplyRowLevelTtlForKeyAndValue() throws Exception {
...
-  }
-
-  @SuppressWarnings("checkstyle:linelength")
-  private static class TtlProcessorSupplier implements ProcessorSupplier<String, String, String, String> {
-
-    @Override
-    public Processor<String, String, String, String> get() {
-      return new TtlDeduplicator();
-    }
+        final var params = ResponsiveKeyValueParams.fact(STORE_NAME).withTtlProvider(
```
Comment (ableegoldman) [inline kafka-client/src/test/java/dev/responsive/kafka/integration/RowLevelTtlIntegrationTest.java:196]: no actual changes to this test at all, just moving some things to the testutils and getting rid of  the duplicate code here

File: kafka-client/src/test/java/dev/responsive/kafka/testutils/Matchers.java [hunk 3]
```diff
@@ -1,4 +1,20 @@
-package dev.responsive.kafka.internal.db.testutils;
```
Comment (ableegoldman) [inline kafka-client/src/test/java/dev/responsive/kafka/testutils/Matchers.java:1]: felt a bit confusing to have multiple testutils packages in different places, and since this one had only the one class in it I decided to just move this class to the main testutils and get rid of this other testutils package.

File: responsive-test-utils/src/test/java/dev/responsive/kafka/api/ResponsiveTopologyTestDriverKeyValueStoreTest.java [hunk 4]
```diff
@@ -38,50 +42,56 @@
...
-    // Then:
-    final List<String> outputs = output.readValuesToList();
-    MatcherAssert.assertThat(outputs, Matchers.contains(
-        "a,100,1,1,alice,CA",
-        "c,102,1,1,alice,CA",
-        "d,103,3,3,carol,CA"
-    ));
-    driver.close();
+    final Topology topology = topology(paramsForType(type));
+    try (final TopologyTestDriver driver = setupDriver(topology)) {
```
Comment (ableegoldman) [inline responsive-test-utils/src/test/java/dev/responsive/kafka/api/ResponsiveTopologyTestDriverKeyValueStoreTest.java:68]: no changes to any of the existing tests in this file, just moved the driver into a try-with-resources

File: responsive-test-utils/src/test/java/dev/responsive/kafka/api/ResponsiveTopologyTestDriverKeyValueStoreTest.java [hunk 5]
```diff
@@ -155,56 +169,104 @@ public void shouldEnforceKeyBasedTtlByAdvancingWallclockTime(final KVSchema type
...
+
+    final ResponsiveKeyValueParams params = paramsForType(type).withTtlProvider(ttlProvider);
+    final Topology topology = deduplicatorApp("input", "output", params);
+    try (final ResponsiveTopologyTestDriver driver = setupDriver(topology)) {
+
+      final TestInputTopic<String, String> inputTopic = driver.createInputTopic(
+          "input",
+          new StringSerializer(),
+          new StringSerializer(),
+          STARTING_TIME,
```
Comment (ableegoldman) [inline responsive-test-utils/src/test/java/dev/responsive/kafka/api/ResponsiveTopologyTestDriverKeyValueStoreTest.java:240]: Once we fix this issue in Streams we should be able to remove this explicit time initialization here and in the driver itself, and have the tests still pass. Until then, we can't verify ttl with millisecond precision because the system time might drift between the driver initialization and the test topic creation

File: kafka-client/build.gradle.kts (hunk 1 above)
Comment (agavra) [reply kafka-client/build.gradle.kts:65]: 🤷 when it comes to build systems, if it works it works

**Review by agavra — APPROVED** [review]
LGTM
